instrumentation/aws-sdk/aws-sdk-2.2/auto/src/main/java/io/opentelemetry/instrumentation/auto/awssdk/v2_2/AwsClientInstrumentation.java

```diff
@@ -17,72 +17,46 @@
 package io.opentelemetry.instrumentation.auto.awssdk.v2_2;
 
 import static io.opentelemetry.javaagent.tooling.ClassLoaderMatcher.hasClassesNamed;
-import static io.opentelemetry.javaagent.tooling.bytebuddy.matcher.AgentElementMatchers.implementsInterface;
-import static net.bytebuddy.matcher.ElementMatchers.isMethod;
-import static net.bytebuddy.matcher.ElementMatchers.isPublic;
-import static net.bytebuddy.matcher.ElementMatchers.nameStartsWith;
 import static net.bytebuddy.matcher.ElementMatchers.named;
 
 import com.google.auto.service.AutoService;
 import io.opentelemetry.javaagent.tooling.Instrumenter;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.Map;
-import net.bytebuddy.asm.Advice;
 import net.bytebuddy.description.method.MethodDescription;
 import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.matcher.ElementMatcher;
-import software.amazon.awssdk.core.client.builder.SdkClientBuilder;
 
-/** AWS SDK v2 instrumentation */
+/**
+ * Injects resource file with reference to our {@link TracingExecutionInterceptor} to allow SDK's
+ * service loading mechanism to pick it up.
+ */
 @AutoService(Instrumenter.class)
-public final class AwsClientInstrumentation extends AbstractAwsClientInstrumentation {
+public class AwsClientInstrumentation extends AbstractAwsClientInstrumentation {
+  @Override
+  public String[] helperResourceNames() {
+    return new String[] {
+      "software/amazon/awssdk/global/handlers/execution.interceptors",
```

Comment:
Without a smoke test, I'm actually a little worried this won't get shaded

Reply:
I'm not following, what do you want shaded here?

Reply:
Sorry nevermind :)

Reply:
😅